cmd/vault-k8s-secret/entrypoint.sh

```
@@ -1,50 +1,103 @@
+echo "Authenticating service account..."
+SA_EMAIL=$(jq -r '.client_email' "${GOOGLE_APPLICATION_CREDENTIALS}")
+SA_KEY=$(jq -r '.private_key' "${GOOGLE_APPLICATION_CREDENTIALS}")
+
+JWT_HEADER=$(printf '{"alg":"RS256","typ":"JWT"}' | openssl base64 -e -A | tr '+/' '-_' | tr -d '=')
+NOW=$(date +%s)
+EXP=$((NOW + 3600))
+JWT_PAYLOAD=$(printf '{"iss":"%s","scope":"https://www.googleapis.com/auth/cloud-platform","aud":"https://oauth2.googleapis.com/token","iat":%d,"exp":%d}' \
+  "${SA_EMAIL}" "${NOW}" "${EXP}" | openssl base64 -e -A | tr '+/' '-_' | tr -d '=')
+
+SIGNING_INPUT="${JWT_HEADER}.${JWT_PAYLOAD}"
+SIGNATURE=$(printf '%s' "${SIGNING_INPUT}" | openssl dgst -sha256 -sign <(printf '%s' "${SA_KEY}") | openssl base64 -e -A | tr '+/' '-_' | tr -d '=')
```

Shellcheck (0.11.0)

[warning] In POSIX sh, process substitution is undefined.

(SC3001)